src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -153,13 +167,20 @@ struct ZoneSet : winrt::implements<ZoneSet, IZoneSet>
     ZoneSetConfig m_config;
 };
 
-IFACEMETHODIMP ZoneSet::AddZone(winrt::com_ptr<IZone> zone) noexcept
+IFACEMETHODIMP ZoneSet::AddZone(winrt::com_ptr<IZone> zone, int id = -1) noexcept
 {
-    m_zones.emplace_back(zone);
-
+    if (id > m_zones.size())
```

Comment:
This is counter-intuitive as it relies on how C++ treats signed/unsigned comparison. When id is negative, it gets converted into an equivalent unsigned value, which is the largest possible unsigned value. Anyway, you will need to drop the additional int parameter, see my other comment.